File: Trailer/SectionHeader.m
#import "SectionHeader.h"

@implementation SectionHeader
{
	CenterTextField *titleView;
}

- (id)initWithTitle:(NSString *)title showRemoveAllButton:(BOOL)show
{
    self = [super initWithFrame:CGRectMake(0, 0, MENU_WIDTH, TITLE_HEIGHT)];
    if (self) {

		self.canDrawSubviewsIntoLayer = YES;

		CGFloat W = MENU_WIDTH-app.scrollBarWidth;
		if(show)
		{
			NSButton *_unpin = [[NSButton alloc] initWithFrame:CGRectMake(W-100, 5.0, 90, TITLE_HEIGHT)];
			[_unpin setTitle:@"Remove All"];
			[_unpin setTarget:self];
			[_unpin setAction:@selector(unPinSelected:)];
			[_unpin setButtonType:NSMomentaryLightButton];
			[_unpin setBezelStyle:NSRoundRectBezelStyle];
			[_unpin setFont:[NSFont systemFontOfSize:10.0]];
			[self addSubview:_unpin];
		}

		NSMutableParagraphStyle *pCenter = [[NSParagraphStyle defaultParagraphStyle] mutableCopy];
		pCenter.alignment = NSCenterTextAlignment;

		NSDictionary *titleAttributes = @{ NSFontAttributeName:[NSFont boldSystemFontOfSize:14.0],
										   NSForegroundColorAttributeName:[COLOR_CLASS controlShadowColor],
										   };
		
		CGRect titleRect = CGRectMake(12, 4.0, W-120-AVATAR_SIZE-LEFTPADDING, TITLE_HEIGHT);
		titleView = [[CenterTextField alloc] initWithFrame:titleRect];
		titleView.attributedStringValue = [[NSAttributedString alloc] initWithString:title
																		  attributes:titleAttributes];
		[self addSubview:titleView];
    }
    return self;
}

- (void)drawRect:(NSRect)dirtyRect
{
	[super drawRect:dirtyRect];
	CGContextRef context = (CGContextRef) [[NSGraphicsContext currentContext] graphicsPort];
	CGContextSetFillColorWithColor(context, [COLOR_CLASS controlShadowColor].CGColor);
	CGFloat offset = [MenuWindow usingVibrancy] ? 2.5 : 3.5;
	CGContextFillRect(context, CGRectMake(1.0, offset, MENU_WIDTH-2.0, 0.5));
}

- (void)unPinSelected:(NSButton *)button
{
	[app sectionHeaderRemoveSelected:titleView.attributedStringValue.string];
}

@end

File: Trailer/SectionHeader.swift
import Cocoa

final class SectionHeader: NSTableRowView {

	var titleView: CenterTextField!

	init(title: String, showRemoveAllButton: Bool) {

		let titleHeight: CGFloat = 42

		super.init(frame: CGRect(x: 0, y: 0, width: MENU_WIDTH, height: titleHeight))

		let W = MENU_WIDTH - app.scrollBarWidth
		if showRemoveAllButton {
			let buttonRect = CGRect(x: W-100, y: 5, width: 90, height: titleHeight)
			let unpin = NSButton(frame: buttonRect)
			unpin.title = "Remove All"
			unpin.target = self
			unpin.action = #selector(unPinSelected)
			unpin.setButtonType(.momentaryLight)
			unpin.bezelStyle = .roundRect
			unpin.font = NSFont.systemFont(ofSize: 10)
			addSubview(unpin)
		}

		let x = W-120-AVATAR_SIZE-LEFTPADDING
		titleView = CenterTextField(frame: CGRect(x: 12, y: 4, width: x, height: titleHeight))
		titleView.attributedStringValue = NSAttributedString(string: title, attributes: [
			NSAttributedString.Key.font: NSFont.boldSystemFont(ofSize: 14),
			NSAttributedString.Key.foregroundColor: NSColor.secondaryLabelColor])
		addSubview(titleView)
	}

	@objc private func unPinSelected() {
		app.removeSelected(on: titleView.attributedStringValue.string)
	}

	required init?(coder: NSCoder) {
		fatalError("init(coder:) has not been implemented")
	}
}
